validation/cubed_sphere_rossby_haurwitz/cubed_sphere_rossby_haurwitz.jl

```diff
@@ -0,0 +1,210 @@
+using Oceananigans.Units
+
+using Oceananigans.Diagnostics: accurate_cell_advection_timescale
+
+#####
+##### Progress monitor
+#####
+
+mutable struct Progress
+    interval_start_time :: Float64
+end
+
+function (p::Progress)(sim)
+    wall_time = (time_ns() - p.interval_start_time) * 1e-9
+    progress = 100 * (sim.model.clock.time / sim.stop_time)
+
+    @info @sprintf("[%06.2f%%] Time: %s, iteration: %d, max(|u⃗|): (%.2e, %.2e) m/s, extrema(η): (min=%.2e, max=%.2e), CFL: %.2e, Δ(wall time): %s",
+                   progress,
+                   prettytime(sim.model.clock.time),
+                   sim.model.clock.iteration,
+                   maximum(abs, sim.model.velocities.u),
+                   maximum(abs, sim.model.velocities.v),
+                   minimum(sim.model.free_surface.η),
+                   maximum(sim.model.free_surface.η),
+                   sim.parameters.cfl(sim.model),
+                   prettytime(wall_time))
+
+    p.interval_start_time = time_ns()
+
+    return nothing
+end
+
+#####
+##### Script starts here
+#####
+
+ENV["DATADEPS_ALWAYS_ACCEPT"] = "true"
+
+Logging.global_logger(OceananigansLogger())
+
+dd32 = DataDep("cubed_sphere_32_grid",
+    "Conformal cubed sphere grid with 32×32 grid points on each face",
+    "https://github.com/CliMA/OceananigansArtifacts.jl/raw/main/cubed_sphere_grids/cubed_sphere_32_grid.jld2",
+    "b1dafe4f9142c59a2166458a2def743cd45b20a4ed3a1ae84ad3a530e1eff538" # sha256sum
+)
+
+dd96 = DataDep("cubed_sphere_96_grid",
+    "Conformal cubed sphere grid with 96×96 grid points on each face",
+    "https://github.com/CliMA/OceananigansArtifacts.jl/raw/main/cubed_sphere_grids/cs96/cubed_sphere_96_grid.jld2"
+)
+
+DataDeps.register(dd32)
+DataDeps.register(dd96)
+
+function diagnose_velocities_from_streamfunction(ψ, grid)
+    ψᶠᶠᶜ = Field(Face, Face,   Center, CPU(), grid, nothing, nothing)
+    uᶠᶜᶜ = Field(Face, Center, Center, CPU(), grid, nothing, nothing)
+    vᶜᶠᶜ = Field(Center, Face, Center, CPU(), grid, nothing, nothing)
+
+    for (f, grid_face) in enumerate(grid.faces)
+        Nx, Ny, Nz = size(grid_face)
+        for i in 1:Nx+1, j in 1:Ny+1
+            ψᶠᶠᶜ.faces[f][i, j, 1] = ψ(grid_face.λᶠᶠᵃ[i, j], grid_face.φᶠᶠᵃ[i, j])
+        end
+
+        for i in 1:Nx+1, j in 1:Ny
+            uᶠᶜᶜ.faces[f][i, j, 1] = (ψᶠᶠᶜ.faces[f][i, j, 1] - ψᶠᶠᶜ.faces[f][i, j+1, 1]) / grid.faces[f].Δyᶠᶜᵃ[i, j]
+        end
+
+        for i in 1:Nx, j in 1:Ny+1
+            vᶜᶠᶜ.faces[f][i, j, 1] = (ψᶠᶠᶜ.faces[f][i+1, j, 1] - ψᶠᶠᶜ.faces[f][i, j, 1]) / grid.faces[f].Δxᶜᶠᵃ[i, j]
+        end
+    end
+
+    return uᶠᶜᶜ, vᶜᶠᶜ, ψᶠᶠᶜ
+end
+
+function cubed_sphere_rossby_haurwitz(grid_filepath)
+
+    ## Grid setup
+
+    H = 8kilometers
+    grid = ConformalCubedSphereGrid(grid_filepath, Nz=1, z=(-H, 0))
+
+    ## Model setup
+
+    model = HydrostaticFreeSurfaceModel(
+              architecture = CPU(),
+                      grid = grid,
+        momentum_advection = VectorInvariant(),
```

Comment:
@ali-ramadhan do we want to have this on here - since we know it is broken?

Reply:
I think?

Reply:
True. I'll revert back to `momentum_advection = nothing` for now until `VectorInvariant()` is correct for the cubed sphere.